app/scripts/metamask-controller.js

```diff
@@ -842,6 +844,28 @@ export default class MetamaskController extends EventEmitter {
     this.publicConfigStore = this.createPublicConfigStore();
   }
 
+  showPopup() {
+    if (this.isUnlocked()) {
+      const chainId = this.networkController.getCurrentChainId();
+      const currentChainId = { chainId }.chainId;
+      const isNetworkUsed = this.appStateController.isNetworkUsed(
+        currentChainId,
+      );
+      const balances = this.cachedBalancesController.getCachedBalancesForCurrentNetwork(
+        currentChainId,
+      );
+      const hasNoFundsOnNetwork = Object.values(balances).indexOf('0x0');
+
+      if (!isNetworkUsed) {
+        this.appStateController.setFirstTimeUsedNetwork(currentChainId);
+
+        if (hasNoFundsOnNetwork !== -1) {
+          this.appStateController.setShowPopup(true);
+        }
+      }
+    }
```

Comment:
[REQ] Moreover I don't think we need this in the background. I'd prefer for the new component to pull in the data from the background (the new usedNetworks state) as well as use the selectors we already have for cachedBalances, etc... and determine if it should return null or render the popover. 

A few reasons why I think we should move this to the UI:
1. Although it has access to all necessary data, the background isn't the best place to manage purely UI state like whether a particular UI element is shown.
2. even the word popup has different connotation in the background context... it could be meaning a new MetaMask popup window for confirmations versus a popup within the UI.

Reply:
Fixed in: 97b7be92fd5d5d4bd4ecd04324f887af3c29b1a5